W-13845340: Update package-mule-application.adoc (#101)
mulesoft/docs-code-builder: modules/ROOT/pages/package-mule-application.adoc

//LOGO (web, desktop, or both)
include::partial$acb-ide-logos.adoc[tags="both-ides"]

> Package Mule application resources in a sharable ZIP file, a deployable JAR file, or a lightweight package that contains.

Reviewer: Seems like the last sentence is incomplete?

Author: revised

> Package Mule application resources in a sharable ZIP file, a deployable JAR file, or a lightweight package that contains.

Author: ```suggestion
Package Mule application resources in a sharable ZIP file, a deployable JAR file that contains sources, or a lightweight package that lacks sources.
```

> Package Mule application resources in a sharable package, a deployable JAR file that contains sources, or a lightweight package that lacks sources.

Author: revised

> Package Mule application resources in a sharable package, a deployable JAR file that contains sources, or a lightweight package that lacks sources.

Author: ```suggestion
Package Mule application resources in a sharable JAR file, a deployable JAR file that contains sources, or a lightweight package that lacks sources.
```

> Package Mule application resources in a sharable ZIP file, a deployable JAR file that contains sources, or a lightweight package that lacks sources.

Author: revised

> Package Mule application resources in a sharable ZIP file, a deployable JAR file that contains sources, or a lightweight package that lacks sources.

Author: ```suggestion
Package Mule application resources in a sharable package, a deployable JAR file that contains sources, or a lightweight package that lacks sources.
```

[[sharable]]
== Package a Project in a Sharable JAR File

> Package source files from your project into a light JAR file that you can share. After you share the JAR file, others can unpack and load into a workspace in their Anypoint Code Builder IDE.

Reviewer: Instead of saying "unpack and load" can we say `Import` and link to the import documentation? Unpack and load maybe lead to people think they can unzip it, and that is not the idea

Author: done

> Package source files from your project into a light JAR file that you can share. After you share the JAR file, others can unpack and load into a workspace in their Anypoint Code Builder IDE.

Author: ```suggestion
Package source files from your project into a light JAR file that you can share. After you share the JAR file, others can import it into a workspace in their Anypoint Code Builder IDE.
```

> Package source files from your project into a light JAR file that you can share. After you share the JAR file, others can import it into a workspace in their Anypoint Code Builder IDE.

Author: revised @pablocabrera85

[INFO] Finished at: 2023-07-21T13:47:56-07:00
[INFO] ------------------------------------------------------------------------
----
+
The name of the JAR file follows this pattern:
+
`_my-project-name_.1.0.0-SNAPSHOT-mule-application.jar`

[[archivable]]
== Package a Lightweight JAR that Lacks Sources

> Package a Mule application without sources. For this lightweight package to deploy successfully at runtime, you must provide dependencies declared in the `pom.xml` file. Otherwise, the deployment will not run. 

Author: revised @pablocabrera85

Reviewer: What do you think of something around:
`Package a Mule application without sources. For this lightweight package to deploy successfully at runtime, the dependencies declared in your project must be accessible to the runtime. Otherwise, the deployment will fail.`
On the other hand we have documented that this does not work. https://docs.mulesoft.com/mule-runtime/4.4/package-a-mule-application
So I'm not sure what is the case.

Author: ```suggestion
Package a Mule application without libraries from the `/lib` directory. To deploy this package successfully at runtime, you must provide dependencies declared in your project, which are not downloaded to the package at runtime. 
```

Author: @pablocabrera85 How about this ☝️ instead? I'm not sure who would use this. I think you were saying that maybe in an on-prem scenario where they were supplied as part of a build system or something?

> Package a Mule application without sources. For this lightweight package to deploy successfully at runtime, you must provide dependencies declared in the `pom.xml` file. Otherwise, the deployment will not run. 

Author: ```suggestion
Package a Mule application without libraries from the `/lib` directory. To deploy this package successfully at runtime, you must provide dependencies, which are not downloaded to the package at runtime. 
```

Author: @pablocabrera85 How about this ☝️ instead? I'm not sure who would use this. I think you were saying that maybe in an on-prem scenario where they were supplied as part of a build system or something?

> Package a Mule application that lacks sources. For this lightweight package to deploy successfully at runtime, you must provide dependencies declared in the `pom.xml` file. Otherwise, the deployment will not run. 

Author: ```suggestion
Package a Mule application without sources. For this lightweight package to deploy successfully at runtime, you must provide dependencies declared in the `pom.xml` file. Otherwise, the deployment will not run. 
```

[INFO] Finished at: 2023-07-21T13:47:56-07:00
[INFO] ------------------------------------------------------------------------
----
+
The name of the JAR file follows this pattern:
+
`_my-project-name_.1.0.0-SNAPSHOT-mule-application.jar`

[[archivable]]

> == Package an Archive 

Reviewer: What is the differences between this tittle and the the package a deployable jar? Is this section for the lightweight package? If that is the case, the app CAN be deployed. Won't include the sources. Deployment will fail if any dependency cannot be resolved by the runtime

Author: revised

Author: @pablocabrera85 How do people usually use this package? What's the intent?

> == Package an Archive 

Author: ```suggestion
== Package a Lightweight JAR that Lacks Sources
```

> == Package a Lightweight JAR that Lacks Sources

Author: revised @pablocabrera85

> Archive Mule application source files and metadata required to import the JAR package back into Anypoint Code Builder.
> 
> This package of a Mule application does not include any dependencies declared in the `pom.xml` file. This JAR file cannot be deployed to a Mule runtime engine. 

Author: ```suggestion
Package a Mule application that lacks sources. For this lightweight package to deploy successfully at runtime, you must provide dependencies declared in the `pom.xml` file. Otherwise, the deployment will not run. 
```
